lib/ask_web/controllers/channel_controller.ex

```diff
@@ -106,4 +106,20 @@ defmodule AskWeb.ChannelController do
 
     render(conn, "show.json", channel: channel |> Repo.preload([:projects, :user]))
   end
+
+  def pause(conn, %{"channel_id" => id}) do
+    channel_params = %{"paused" => true}
+    result = AskWeb.ChannelController.update(conn, %{"id" => id, "channel" => channel_params})
```

Comment:
Controller functions shouldn't call other controller functions.

This `update` endpoint may (now or later), for example, have different permissions checks than the ones needed for this `pause` endpoint - and coupling them won't help.